test/cli/index_test.cpp

```diff
@@ -0,0 +1,133 @@
+#include <fstream>
+#include <string>
+
+#include "cli_test.hpp"
+
+struct index_test : public cli_test
+{
+    void run_index_test(std::string const & index_command,
+                        std::string const & db_file,
+                        std::string const & index_file,
+                        std::string const & index_type,
+                        std::string const & reduction,
+                        std::string const & control_file)
+    {
+        cli_test_result result = execute_app("lambda3", index_command,
+                                             "-d", data(db_file),
+                                             "-i", index_file,
+                                             "--db-index-type", index_type,
+                                             "-r", reduction);
+
+        ASSERT_EQ(result.exit_code, 0);
+
+        std::system(("md5sum " + index_file + " > md5sum_test.txt").c_str());
+        std::system(("md5sum " + (std::string) data(control_file) + " > md5sum_control.txt").c_str());
+
+        std::ifstream test_output ("md5sum_test.txt");
+        std::ifstream control_output ("md5sum_control.txt");
+
+        std::string test_line;
+        std::string control_line;
+
+        getline(test_output, test_line);
+        getline(control_output, control_line);
+
+        std::string test_sum = test_line.substr(0, test_line.find(' '));
+        std::string control_sum = control_line.substr(0, control_line.find(' '));
+
+        ASSERT_EQ(test_sum, control_sum);
+    }
+};
+
+TEST_F(index_test, no_options)
+{
+    cli_test_result result = execute_app("lambda3");
+    std::string expected
+    {
+        "lambda3 - Lambda, the Local Aligner for Massive Biological DatA.\n"
+        "================================================================\n"
+        "    [OPTIONS] COMMAND [COMMAND-OPTIONS]\n"
+        "    Try -h or --help for more information.\n"
+    };
+    ASSERT_EQ(result.exit_code, 0);
+    ASSERT_EQ(result.out, expected);
+    ASSERT_EQ(result.err, std::string{});
```

Comment:
In these cases, I would use `EXPECT_EQ` and not `ASSERT_EQ` as the subsequent steps are valid (and interesting!) even if one of them fails.

Reply:
Yes, I only changed it to see the error messages when the test was not running through. Will revert that.